client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_POSTER_UPDATE } from 'state/action-types';
+import {
+	updateVideoEditorPosterSuccess,
+	updateVideoEditorPosterFailure,
+	updatingVideoEditorPoster,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updateVideoEditorPoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params ) && ! ( 'at_time' in action.params ) ) {
+		return next( action );
+	}
+
+	dispatch( updatingVideoEditorPoster() );
+
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'POST',
+		path: `/videos/${ action.videoId }/poster`,
+		body: 'at_time' in action.params ? action.params : undefined,
+		formData: 'file' in action.params ? [ [ 'poster', action.params.file ] ] : undefined,
+		onSuccess: action,
+		onFailure: action,
+	} ) );
+
+	return next( action );
```

Comment:
@dmsnell is there ever a case where we wouldn't want to pass the action along the middleware chain? As is we'll get weird oddities like the dispatch action not being recorded by redux dev tools if we omit `return next( action )` in addition to missing any other side effect middleware that was lower on the chain.

For example in console if you dispatch:
```
dispatch( { type: 'ACCOUNT_RECOVERY_RESET_REQUEST' } );
```
We'll see the other actions recorded but not the initial action request :
![screen shot 2017-03-10 at 1 51 54 pm](https://cloud.githubusercontent.com/assets/1270189/23814588/efea2b64-0598-11e7-8cb6-a5f3d9fc36be.png)
https://github.com/Automattic/wp-calypso/pull/11699/files#diff-2c98cce53eb8b5257f1bbca88632820cR17
cc @southp